docs/guides/code_examples/crawler_custom_parser/selectolax_crawler.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from selectolax.lexbor import LexborHTMLParser, LexborNode
+
+from crawlee.crawlers import AbstractHttpCrawler, HttpCrawlerOptions
+
+from .selectolax_context import SelectolaxLexborContext
+from .selectolax_parser import SelectolaxLexborParser
+
+if TYPE_CHECKING:
+    from collections.abc import AsyncGenerator
+
+    from typing_extensions import Unpack
+
+    from crawlee.crawlers._abstract_http import ParsedHttpCrawlingContext
+
+
+class SelectolaxLexborCrawler(
+    AbstractHttpCrawler[SelectolaxLexborContext, LexborHTMLParser, LexborNode]
+):
+    """Custom crawler using Selectolax Lexbor for HTML parsing."""
+
+    def __init__(
+        self,
+        **kwargs: Unpack[HttpCrawlerOptions[SelectolaxLexborContext]],
+    ) -> None:
+        # Final step converts the base context to custom context type.
+        async def final_step(
+            context: ParsedHttpCrawlingContext[LexborHTMLParser],
+        ) -> AsyncGenerator[SelectolaxLexborContext, None]:
+            yield SelectolaxLexborContext.from_parsed_http_crawling_context(context)
+
+        # Build context pipeline: HTTP request -> parsing -> custom context.
+        kwargs['_context_pipeline'] = (
+            self._create_static_content_crawler_pipeline().compose(final_step)
+        )
```

Comment:
It'd be nice to mention that these are not strictly necessary, you only need this if you want "idiomatic" helpers on the context (`parser` in this case). If you don't mind sticking with `parsed_content`, you can skip a lot of boilerplate.